### PR TITLE
Add generic_hygrostat integration

### PR DESCRIPTION
**Pull request by Shulyaka**
## Proposed change

Add `generic_hygrostat` integration

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [x] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: home-assistant/core#36759
- Link to parent pull request in the Brands repository: home-assistant/brands#1726
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Review by maxiangelo — DISMISSED** [review]
Looks good!

**Comment by barberio** [comment]
Please specify the unit of time used in min_cycle_duration. Seconds, minutes, milliseconds?

File: source/_integrations/generic_hygrostat.markdown [hunk 1]
```diff
@@ -0,0 +1,115 @@
+---
+title: Generic Hygrostat
+description: Virtual hygrostat device
+ha_category:
+  - Humidifier
+ha_release: "0.112"
```
Comment (frenck) [inline source/_integrations/generic_hygrostat.markdown]:
```suggestion
ha_release: 2021.8
```

File: source/_integrations/generic_hygrostat.markdown [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+ha_domain: generic_hygrostat
+ha_quality_scale: internal
+---
+
+The `generic_hygrostat` humidifier platform is a virtual hygrostat implemented in Home Assistant. It uses a sensor and a switch connected to a humidifier or dehumidifier under the hood. When in humidifier mode, if the measured humidity is less than the target humidity, the humidifier will be turned on and turned off when the required humidity is reached. When in dehumidifier mode, if the measured humidity is greater than the target humidity, the dehumidifier will be turned on and turned off when required humidity is reached. One Generic Hygrostat entity can only control one switch. If you need to activate two switches, one for a humidifier and one for a dehumidifier, you will need two Generic Hygrostat entities.
```
Comment (frenck) [inline source/_integrations/generic_hygrostat.markdown]:
```suggestion
The `generic_hygrostat` humidifier integration is a virtual hygrostat implemented in Home Assistant. It uses a sensor and a switch connected to a humidifier or dehumidifier under the hood. When in humidifier mode, if the measured humidity is less than the target humidity, the humidifier will be turned on and turned off when the required humidity is reached. When in dehumidifier mode, if the measured humidity is greater than the target humidity, the dehumidifier will be turned on and turned off when required humidity is reached. One Generic Hygrostat entity can only control one switch. If you need to activate two switches, one for a humidifier and one for a dehumidifier, you will need two Generic Hygrostat entities.
```

File: source/_integrations/generic_hygrostat.markdown [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  required: false
+  default: 0
+  type: integer
+max_humidity:
+  description: Set maximum set point available.
+  required: false
+  default: 100
+  type: integer
+target_humidity:
+  description: Set initial target temperature. This value will be used as fallback when previous setpoint is not available.
```
Comment (frenck) [inline source/_integrations/generic_hygrostat.markdown]:
```suggestion
  description: Set initial target temperature. This value will be used as a fallback when the previous setpoint is not available.
```

**Review by frenck — APPROVED** [review]

**Pull request merged**